### ISSUE TITLE
ENH, MAINT: perform finetuning/hyperparameter optimization for opencv and sam2 detection modules

### ISSUE THREAD

**Issue by adamwitmer** [lanl/ldrd_neat_ml#13]
For open PR's #2 and #3, the lack of fine-tuning and/or systematic hyperparameter optimization for the `opencv` `SimpleBlobDetector` and `SAM2` models/parameters presents a major limitation of the current work. These procedures should be performed to guarantee optimal performance of the respective segmentation algorithms, ideally in relation to some ground-truth image(s) by using i.e. `intersection-over-union (IoU)` aka `Jaccard score` (https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_score.html) as a metric for quantifying segmentation performance.

For `SimpleBlobDetector`, the model parameters are described here: https://opencv.org/blog/blob-detection-using-opencv/, and are currently determined using manual tuning based on visual inspection. Similarly, the current implementation of `SAM2` algorithm uses pre-trained weights and does not implement any fine-tuning of the model on the "raw" microscopy image data or systematic hyperparameter optimization for the `SAM2AutomaticMaskGenerator` parameters as described in the source code: https://github.com/facebookresearch/sam2/blob/2b90b9f5ceec907a1c18123530e92e794ad901a4/sam2/automatic_mask_generator.py#L36
